### PR TITLE
Added tests for MmAllocateSystemMemory and MmFreeSystemMemory

### PR DESCRIPTION
**Pull request by ergo720**
Note that I have only confirmed these tests with [nboxkrnl](https://github.com/ergo720/nboxkrnl). So, before merging this, someone should probably check that they work as expected on a real xbox too.

### CODE REVIEW

**Comment by ergo720** [comment]
After a lengthy discussion with @RadWolfie,

File: src/tests/mm/MmAllocateSystemMemory.c [hunk 1]
```diff
@@ -1,8 +1,58 @@
...
 
 void test_MmAllocateSystemMemory()
 {
-    /* FIXME: This is a stub! implement this function! */
+    const char* func_num = "0x00A7";
+    const char* func_name = "MmAllocateSystemMemory";
+    BOOL test_passed = 1;
+    print_test_header(func_num, func_name);
+
+   // NOTE: these tests will be marked as failed if not enough memory can be allocated for them
```
Comment (RadWolfie) [inline src/tests/mm/MmAllocateSystemMemory.c]:
```suggestion
   // NOTE: these tests could be marked as failed if not enough memory can be allocated for them
```

File: src/tests/mm/MmFreeSystemMemory.c [hunk 2]
```diff
@@ -1,8 +1,34 @@
...
+
+    // Test #1 - free one page
+    PVOID sys_addr = MmAllocateSystemMemory(PAGE_SIZE, PAGE_READONLY);
+    GEN_CHECK(sys_addr != NULL, TRUE, "SystemMemoryAddress (RO - PAGE_SIZE)");
+    if (sys_addr) {
+        num_of_pages = MmFreeSystemMemory(sys_addr, 0);
+        GEN_CHECK(num_of_pages, 1, "num_of_pages (RO - PAGE_SIZE)");
+    }
+
+    // Test #1 - free two pages
```
Comment (RadWolfie) [inline src/tests/mm/MmFreeSystemMemory.c]:
```suggestion
    // Test #2 - free two pages
```

File: src/tests/mm/MmAllocateSystemMemory.c (hunk 1 above)
Comment (RadWolfie) [reply src/tests/mm/MmAllocateSystemMemory.c]: For clarification, if memory allocated failed. The following reasons could be the cause:
- Not enough memory to allocated (previous xbe used up all the memory and didn't freed them up)
- defective chipset (hardware issue)
- stub function which only return null pointer. (custom kernel issue, not an issue with the test itself)
- etc

**Comment by PatrickvL** [comment]
Why was this closed without merging?

**Comment by ergo720** [comment]
> Why was this closed without merging?

Because I and RadWolfie had different views on the PR, and we couldn't agree on it.

**Comment by PatrickvL** [comment]
Sad

**Pull request closed without merging**